File: app/src/main/java/app/we/go/emojidraw/arch/di/ApplicationModule.java
package app.we.go.emojidraw.arch.di;

import android.content.Context;

import javax.inject.Singleton;

import app.we.go.emojidraw.R;
import app.we.go.emojidraw.arch.di.qualifier.CanvasSize;
import dagger.Module;
import dagger.Provides;

/**
 * Provides general, cross-application dependencies
 */
@Module
public class ApplicationModule {

    @Provides
    @Singleton
    @CanvasSize
    Integer provideCanvasSize(Context context) {
        return context.getResources().getDimensionPixelSize(R.dimen.drawing_area);
    }


}

File: app/src/main/java/app/we/go/emojidraw/arch/di/ApplicationModule.kt
package app.we.go.emojidraw.arch.di

import android.content.Context
import app.we.go.emojidraw.R
import app.we.go.emojidraw.arch.di.qualifier.CanvasSize
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ApplicationModule(private val context: Context) {


    @Provides
    @Singleton
    fun provideContext(): Context {
        return context
    }



    @Provides
    @Singleton
    @CanvasSize
    fun provideCanvasSize(): Int {
        return context.resources.getDimensionPixelSize(R.dimen.drawing_area)
    }

}